odoo/models.py

```diff
@@ -5373,11 +5373,11 @@ def filtered_domain(self, domain):
                 if comparator in ('like', 'ilike', '=like', '=ilike', 'not ilike', 'not like'):
                     value_esc = value.replace('_', '?').replace('%', '*').replace('[', '?')
                 records_ids = OrderedSet()
+                if comparator in ('child_of', 'parent_of'):
+                    result.append(self & self.search([d]))
+                    continue
                 for rec in self:
                     data = rec.mapped(key)
-                    if comparator in ('child_of', 'parent_of'):
-                        value = data.search([(data._parent_name, comparator, value)]).ids
```

Comment:
This search actually feels wrong to me: that's not what the condition means.  IMHO it instead should be
```python
value = data.search([('id', comparator, value)]).ids
```